notification/slack/usergroupsender.go

```diff
@@ -132,27 +131,31 @@ func (s *UserGroupSender) Send(ctx context.Context, msg notification.Message) (*
 		log.Log(log.WithField(ctx, "SlackUGErrorID", errID), err)
 		var buf bytes.Buffer
 		err := userGroupErrorUpdate.Execute(&buf, userGroupError{
-			ErrorID: errID,
-			GroupID: ugID,
-			OnCall:  onCallMsg,
+			ErrorID:  errID,
+			GroupID:  ugID,
+			ErrorMsg: errorMsg,
 		})
 		if err != nil {
 			return nil, fmt.Errorf("execute template: %w", err)
 		}
-		onCallMsg = buf.String()
+		errorMsg = buf.String()
 		stateDetails = "failed to update user-group, sent error to channel and log"
 	}
 
 	var ts string
-	err = s.withClient(ctx, func(c *slack.Client) error {
-		_, ts, err = c.PostMessageContext(ctx, chanID, slack.MsgOptionText(onCallMsg, false))
+
+	// Only send to the channel if an error occurred
+	if stateDetails != "" {
+		err = s.withClient(ctx, func(c *slack.Client) error {
+			_, ts, err = c.PostMessageContext(ctx, chanID, slack.MsgOptionText(errorMsg, false))
+			if err != nil {
+				return fmt.Errorf("post message to channel '%s': %w", chanID, err)
+			}
+			return nil
+		})
 		if err != nil {
```

Comment:
Let's flatten this.
https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

We can invert the check for the quick/easy return (since there's no state details or external ID). Secondly, we can just return the error to `withClient` and add the context in the external error check.
```go
if stateDetails == "" {
	return &notification.SentMessage{State: notification.StateDelivered}, nil
}

err = s.withClient(ctx, //...
	_, ts, err = c.PostMessageContext( //...
	return err
})
if err != nil {
	return nil, fmt.Errorf("post message to channel '%s': %w", chanID, err)
}
```